src/Playgrounds/CosmosEndToEnd/CosmosEndToEnd.ApiService/Program.cs

```diff
@@ -0,0 +1,70 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text.Json.Serialization;
+using CosmosEndToEnd.ApiService;
+using Microsoft.Azure.Cosmos;
+
+var sessionId = Guid.NewGuid().ToString();
+
+var builder = WebApplication.CreateBuilder(args);
+
+builder.AddServiceDefaults();
+builder.AddAzureCosmosDB("db",
+                         settings =>
+                            {
+                                settings.IgnoreEmulatorCertificate = true;
+                            },
+                         clientOptions =>
+                         {
+                             // Default serializer for Cosmos V3 client is JSON.NET, this changes
+                             // us to use S.T.J for this playground.
+                             clientOptions.Serializer = new StjSerializer(new System.Text.Json.JsonSerializerOptions());
+                         });
+
+builder.Services.AddKeyedSingleton<Container>("entries", (sp, _) =>
+{
+    var db = sp.GetRequiredService<Database>();
+    var container = db.CreateContainerIfNotExistsAsync("entries", "/sessionId").Result.Container;
```

Comment:
Is there a way to avoid the blocking call?

Reply:
This is occurring in DI wire up which isn't async. It may be that we just don't do this database injection, and we leave all this to the developer, but my goal was to try and make it as seamless as possible. But perhaps this should be a Cosmos SDK responsibility as @eerhardt mentions above.